src/promptflow/promptflow/executor/_dag_manager.py

```diff
@@ -44,14 +45,17 @@ def pop_bypassable_nodes(self) -> List[Node]:
             del self._pending_nodes[node.name]
         return bypassed_nodes
 
-    def get_node_valid_inputs(self, node: Node) -> Mapping[str, Any]:
+    def get_node_valid_inputs(self, node: Node, f: Callable) -> Mapping[str, Any]:
         """Returns the valid inputs for the node, including the flow inputs, literal values and
         the outputs of completed nodes."""
-        return {
-            name: self._get_node_dependency_value(i)
-            for name, i in (node.inputs or {}).items()
-            if not self._is_node_dependency_bypassed(i)
-        }
+        result = {}
+        signature = inspect.signature(f).parameters
+        for name, i in (node.inputs or {}).items():
+            if self._get_node_dependency_value(i) is None and signature[name].default is not inspect.Parameter.empty:
+                result[name] = signature[name].default
```

Comment:
We'd better not to set the default value here, the recommended way to apply default value is not passing the keys to the function, as what we did before.

Reply:
The logic of the current code is passing None to the function when the node is bypassed and the function has no default value, and ignoring the output when the node is bypassed and the function has default value.